packages/react-data-grid/src/masks/InteractionMasks.tsx

```diff
@@ -0,0 +1,750 @@
+      cursorCell: cellPosition,
+      isDragging
+    };
+  }
+
+  onSelectCellRangeStarted = (selectedPosition: Position): void => {
+    this.setState({
+      selectedRange: this.createSingleCellSelectedRange(selectedPosition, true),
+      selectedPosition
+    }, () => {
+      if (this.props.onCellRangeSelectionStarted) {
+        this.props.onCellRangeSelectionStarted(this.state.selectedRange);
+      }
+    });
+  };
+
+  onSelectCellRangeUpdated = (cellPosition: Position, isFromKeyboard: boolean, callback?: () => void): void => {
+    if (!this.state.selectedRange.isDragging && !isFromKeyboard || !this.isCellWithinBounds(cellPosition)) {
+      return;
+    }
+
+    const startCell = this.state.selectedRange.startCell || this.state.selectedPosition;
+    const colIdxs = [startCell.idx, cellPosition.idx].sort((a, b) => a - b);
+    const rowIdxs = [startCell.rowIdx, cellPosition.rowIdx].sort((a, b) => a - b);
+    const topLeft: Position = { idx: colIdxs[0], rowIdx: rowIdxs[0] };
+    const bottomRight: Position = { idx: colIdxs[1], rowIdx: rowIdxs[1] };
+
+    const selectedRange: SelectedRange = {
+      ...this.state.selectedRange,
+      // default the startCell to the selected cell, in case we've just started via keyboard
+      startCell: this.state.selectedRange.startCell || this.state.selectedPosition,
+      // assign the new state - the bounds of the range, and the new cursor cell
+      topLeft,
+      bottomRight,
+      cursorCell: cellPosition
+    };
+
+    this.setState({
+      selectedRange
+    }, () => {
+      if (this.props.onCellRangeSelectionUpdated) {
+        this.props.onCellRangeSelectionUpdated(this.state.selectedRange);
+      }
+      if (callback) {
+        callback();
+      }
+    });
+  };
+
+  onSelectCellRangeEnded = (): void => {
+    const selectedRange = { ...this.state.selectedRange, isDragging: false };
+    this.setState({ selectedRange }, () => {
+      if (this.props.onCellRangeSelectionCompleted) {
+        this.props.onCellRangeSelectionCompleted(this.state.selectedRange);
+      }
+
+      // Focus the InteractionMasks, so it can receive keyboard events
+      this.focus();
+    });
+  };
+
+  isDragEnabled(): boolean {
+    return this.isSelectedCellEditable();
+  }
+
+  handleDragStart = (e: React.DragEvent<HTMLDivElement>): void => {
+    const { selectedPosition } = this.state;
+    // To prevent dragging down/up when reordering rows. (TODO: is this required)
+    if (selectedPosition.idx > -1) {
+      e.dataTransfer.effectAllowed = 'copy';
+      // Setting data is required to make an element draggable in FF
+      const transferData = JSON.stringify(selectedPosition);
+      try {
+        e.dataTransfer.setData('text/plain', transferData);
+      } catch (ex) {
+        // IE only supports 'text' and 'URL' for the 'type' argument
+        e.dataTransfer.setData('text', transferData);
+      }
+      this.setState({
+        draggedPosition: selectedPosition
```

Comment:
It looks like we can initialize `overRowIdx` as `selectedPosition.rowIdx` as we return null when `overRowIdx === rowIdx`

Reply:
Done.